bin/tests/common.rs

```diff
@@ -87,7 +87,22 @@ pub fn truncate_file(file_path: &PathBuf) -> Result<(), std::io::Error> {
     Ok(())
 }
 
-pub fn spawn_agent(dir_path: &str) -> Child {
+#[derive(Default)]
+pub struct AgentSettings<'a> {
+    pub log_dirs: &'a str,
+    pub exclusion_regex: &'a str,
```

Comment:
Minor nit: It's a bit more idiomatic to use an Option for fields that may or not be present, rather than checking for the empty str

Reply:
makes sense, I'll change it.